var/spack/repos/builtin/packages/py-python-lsp-server/package.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyPythonLspServer(PythonPackage):
+    """A Python 3.7+ implementation of the Language Server Protocol."""
+
+    homepage = "https://github.com/python-lsp/python-lsp-server"
+    pypi = "python-lsp-server/python-lsp-server-1.6.0.tar.gz"
+
+    maintainers = ["alecbcs"]
+
+    version("1.6.0", sha256="d75cdff9027c4212e5b9e861e9a0219219c8e2c69508d9f24949951dabd0dc1b")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools@61.2.0:", type="build")
```

Comment:
Missing build dep: `setuptools_scm[toml]>=3.4.3`

Reply:
Oh good catch! I'll make sure to update and add this dep! By the way what is `setuptools_scm[toml]`? I've seen this as part of a couple of packages but haven't ever used it myself.

Reply:
This would be `py-setuptools-scm@3.4.3:+toml`. Pip's extras (brackets) are equivalent to Spack's variants (plus/minus).

Reply:
Oh very cool! Thanks for the explanation.